tests/phpunit/integration/Modules/Analytics_4/Synchronize_PropertyTest.php

```diff
@@ -0,0 +1,225 @@
+use Google\Site_Kit\Tests\TestCase;
+use Google\Site_Kit\Tests\FakeHttp;
+use Google\Site_Kit_Dependencies\GuzzleHttp\Psr7\Request;
+use Google\Site_Kit_Dependencies\GuzzleHttp\Psr7\Response;
+use Google\Site_Kit_Dependencies\Google\Service\GoogleAnalyticsAdmin\GoogleAnalyticsAdminV1betaProperty;
+
+/**
+ * @group Modules
+ * @group Analytics_4
+ */
+class Synchronize_PropertyTest extends TestCase {
+
+	/**
+	 * @var Synchronize_Property
+	 */
+	protected $synchronize_property;
+
+	/**
+	 * @var Analytics_4
+	 */
+	protected $analytics_4;
+
+	/**
+	 * @var User_Options
+	 */
+	protected $user_options;
+
+	/**
+	 * @var Options
+	 */
+	protected $options;
+
+	/**
+	 * @var Authentication
+	 */
+	protected $authentication;
+
+	public function set_up() {
+		parent::set_up();
+
+		$user = self::factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user->ID );
+
+		$context              = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options        = new Options( $context );
+		$this->user_options   = new User_Options( $context );
+		$this->authentication = new Authentication( $context, $this->options, $this->user_options );
+		$modules              = new Modules( $context, $this->options, $this->user_options, $this->authentication );
+		$dismissed_items      = new Dismissed_Items( $this->user_options );
+		$encrypted_options    = new Encrypted_Options( $this->options );
+		$permissions          = new Permissions( $context, $this->authentication, $modules, $this->user_options, $dismissed_items );
+
+		// Fake a valid authentication token on the client.
+		$this->authentication->get_oauth_client()->set_token( array( 'access_token' => 'valid-auth-token' ) );
+		$this->authentication->verification()->set( true );
+
+		$encrypted_options->set(
+			Credentials::OPTION,
+			array(
+				'oauth2_client_id'     => 'test-client-id',
+				'oauth2_client_secret' => 'test-client-secret',
+			)
+		);
+		add_filter( 'googlesitekit_setup_complete', '__return_true', 100 );
+
+		$permissions->register();
+
+		$this->analytics_4 = new Analytics_4(
+			$context,
+			$this->options,
+			$this->user_options,
+			$this->authentication
+		);
+
+		$this->set_analytics_settings();
+
+		$this->synchronize_property = new Synchronize_Property(
+			$this->analytics_4,
+			$this->user_options,
+			$this->authentication
+		);
+	}
+
+	public function set_analytics_settings() {
+		// Ensure Anlytics 4 module is connected
+		$this->options->set(
+			Analytics_4_Settings::OPTION,
+			array(
+				'accountID'       => '12345678',
+				'propertyID'      => '987654321',
+				'webDataStreamID' => '1234567890',
+				'measurementID'   => 'G-A1B2C3D4E5',
+			)
+		);
+
+		// Set owner ID, using base analytics settings.
+		$this->options->set(
+			Settings::OPTION,
+			array(
+				'ownerID' => get_current_user_id(),
+			)
+		);
+	}
+
+	public function fake_analytics_4_property_object( $property_id, $create_time ) {
```

Comment:
This method could be a bit more expressive about what it's doing. It's a little ambiguous as it is. Perhaps something like this
```suggestion
	public function fake_analytics_4_property_response( $property_id, $create_time ) {
```